cranelift/codegen/src/isa/aarch64/lower/isle.rs

```diff
@@ -286,4 +286,105 @@ where
         let amount = val.value() & u8::try_from(ty.bits() - 1).unwrap();
         ImmShift::maybe_from_u64(u64::from(ty.bits()) - u64::from(amount)).unwrap()
     }
+
+    fn icmp_cond_zero(&mut self, cond: &IntCC) -> Option<IntCC> {
+        match cond {
+            &IntCC::Equal => Some(IntCC::Equal),
+            &IntCC::SignedGreaterThanOrEqual => Some(IntCC::SignedGreaterThanOrEqual),
+            &IntCC::SignedGreaterThan => Some(IntCC::SignedGreaterThan),
+            &IntCC::SignedLessThanOrEqual => Some(IntCC::SignedLessThanOrEqual),
+            &IntCC::SignedLessThan => Some(IntCC::SignedLessThan),
+            _ => None,
+        }
+    }
+
+    fn fcmp_cond_zero(&mut self, cond: &FloatCC) -> Option<FloatCC> {
+        match cond {
+            &FloatCC::Equal => Some(FloatCC::Equal),
+            &FloatCC::GreaterThanOrEqual => Some(FloatCC::GreaterThanOrEqual),
+            &FloatCC::GreaterThan => Some(FloatCC::GreaterThan),
+            &FloatCC::LessThanOrEqual => Some(FloatCC::LessThanOrEqual),
+            &FloatCC::LessThan => Some(FloatCC::LessThan),
+            _ => None,
+        }
+    }
+
+    fn fcmp_cond_notEq(&mut self, cond: &FloatCC) -> Option<FloatCC> {
+        match cond {
+            &FloatCC::NotEqual => Some(FloatCC::NotEqual),
+            _ => None,
+        }
+    }
+
+    fn icmp_cond_notEq(&mut self, cond: &IntCC) -> Option<IntCC> {
+        match cond {
+            &IntCC::NotEqual => Some(IntCC::NotEqual),
+            _ => None,
+        }
+    }
+
+    fn fcompare(&mut self, cond: &FloatCC) -> VecMisc2 {
```

Comment:
a more descriptive name might be good for this helper: `float_cc_cmp_zero_to_vec_misc_op` maybe? `fcompare` doesn't tell the reader what it does, other than suggest perhaps it returns an instruction or something to do a compare... also, the name does not at all suggest the connection to zero.